migrations/1729687489331-move-applications-custom-templates.ts

```diff
@@ -0,0 +1,129 @@
+import { Application, User } from '@models';
+import { startDatabaseForMigration } from '../src/utils/migrations/database.helper';
+import { CustomTemplate } from '@models/customTemplate.model';
+/**
+ * Sample function of up migration
+ *
+ * @returns just migrate data.
+ */
+export const up = async () => {
```

Comment:
@morgang-adappt 

the migrations are fine to create the objects but they don't link the new objects to the places they are currently used, so the applications would still be broken after running them

in addition, we should query the dashboard pages & workflow dashboard steps where the current template / DL is being used, and replace the ID with the new ID, or, make sure that the ID you use when creating the custom template / distribution list is the one that is currently being used ( that's certainly easier ) so the applications are working after running the migrations

Reply:
in addition, the last step of the migrations should certainly be a edition of the application itself, to remove the previous field value ( as it's useless once the migration is done )

Reply:
also, I believe the logic to fetch an existing DL ( and same for template ), by its name, and then, adding suffix "_new" to it doesn't work and can potentially create a duplication issue

indeed, if you already have a distribution list ( or template ), which has the name of the current one you're applied the logic to + "_new" suffix, then when saving it, you'll face a duplication issue, as the new name you automatically set already exist, so the logic should be something that uses regex, so you can also search for DLs or templates with current name or, starting with current name & suffix _new ( so it can you have _new_1, _new_2 etc it also works